reconcile/utils/oc_map.py

```diff
@@ -0,0 +1,317 @@
+import logging
+from threading import Lock
+from typing import (
+    Any,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Protocol,
+    Sequence,
+    Union,
+)
+
+from sretoolbox.utils import threaded
+
+from reconcile.utils.oc import (
+    OC,
+    JumpHostSSH,
+    OCDeprecated,
+    OCLogMsg,
+    StatusCodeError,
+)
+from reconcile.utils.secret_reader import (
+    HasSecret,
+    SecretNotFound,
+    SecretReaderBase,
+)
+
+
+class HasDisable(Protocol):
+    integrations: Optional[list[str]]
+    e2e_tests: Optional[list[str]]
+
+
+class HasCluster(Protocol):
```

Comment:
i'm always having a hard time with python Protocol naming, especially with the `Has` prefix. i had a discussion with @chassing about that. in a nutshell: in a situation like this, the protocol describes an object "being" something, and therefore it would be ok to bluntly naming it `Cluster` or `ClusterConnectionConfig`. we can see this approach a lot within the collection Protocols, e.g. `Mapping` or `Sequence`.

Reply:
I still had our old convention in mind. But what you suggest makes sense - TIL :) :+1: 

Anyways I'm more and more in favor of using `ClusterParameters` instead of Protocols. Now since you mentioned above that you like it too, I will introduce `ClusterParameters`.

Reply:
cool. thanks!

Reply:
Introduced now `OCConnectionParameters` class. It still has Protocols for convenience purposes, but they are named now as you suggested.